src/simplewallet/simplewallet.cpp

```diff
@@ -2188,6 +2189,28 @@ bool simple_wallet::set_ignore_fractional_outputs(const std::vector<std::string>
   return true;
 }
 
+bool simple_wallet::set_device_name(const std::vector<std::string> &args/* = std::vector<std::string>()*/)
+{
+  const auto pwd_container = get_and_verify_password();
```

Comment:
Hmm you are right, the on the fly change of the device is not really needed.

The original idea is to address the situation where the device descriptor changes (device path) so the user can fix the device name / path, otherwise wallet won't connect / work.
This solution does not fix it well as the wallet cannot be started if the device is not found. I will just let to specify the device with the command line argument, that should do the job. Thanks!